src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -26,6 +26,8 @@ def create_storage_account(cmd, resource_group_name, account_name, sku=None, loc
         params.identity = Identity()
     if https_only:
         params.enable_https_traffic_only = https_only
+    # temporary fix to allow idempotent create when value is None. (sdk defaults with False)
+    params.is_hns_enabled = None
```

Comment:
Curious why can this parameter not be set?  Also, what API version of storage has this?  Does the stack version of storage have this?

Reply:
The parameter for the storage service is immutable. The problems occurs when an existing account has the parameter as `null` but then tries to create again(PUT) with a false bool. This fails even though the command should be idempotent.

It affects the newer api-versions, only available in latest. For stack/hybrid, since the parameter is not in the sdk's method signature, it won't be sent along.